src/infiniteLoops/InfiniteLoops.ts

```diff
@@ -0,0 +1,278 @@
+import * as es from 'estree'
+import { Value } from '../types'
+import { evaluateBinaryExpression, evaluateUnaryExpression } from '../utils/operators'
+import * as stype from './symTypes'
+import { checkBinaryExpression, checkUnaryExpression } from '../utils/rttc'
+import { serialize } from './serializer'
+import { getFirstCall, symbolicExecute } from './symbolicExecutor'
+import { getCheckers } from './analyzer'
+import * as create from '../utils/astCreator'
+
+interface SimpleEnv {
+  constants: [string, any][]
+  tset: stype.TransitionSet
+  chapter2: boolean
+}
+function newEnv(chap?: boolean) {
+  return { constants: [], tset: new Map(), chapter2: chap } as SimpleEnv
+}
+
+function getVariable(envs: SimpleEnv[], name: string) {
+  for (const env of envs) {
+    for (const [key, val] of env.constants) {
+      if (key === name) {
+        return val
+      }
+    }
+  }
+  return undefined
+}
+
+function setVariable(envs: SimpleEnv[], name: string, value: any) {
+  envs[0].constants.unshift([name, value])
+  return undefined
+}
+
+function getConsts(envs: SimpleEnv[]) {
+  const encountered: string[] = []
```

Comment:
It's supposed to keep track of all the variables encountered in the innermost scope so we don't duplicate them in the results list. My mistake, I forgot to add the line that populates it.

I added the line I missed.

(Right now it works fine coincidentally because of the way getStore (the only function that uses getConsts) is implemented.)